File: 2019/day1c/day1.py
import math

def simple_math(value):
    value = value /3
    value = math.floor(value)
    value = value-2
    print(value)
    return value

def complex_math(old_value):
    new_value = []
    temp_value=old_value
    value= 0
    if (old_value >= 0 ):
        for i in new_value:
            value += new_value[i]
        value = value + temp_value 
    else:
            new_value = simple_math(old_value)
            old_value -= new_value 
            print(old_value)

    return value 



def main():
    maths=0
    with open("test.txt","r") as t:
        for line in t:
            sm = simple_math(int(line))
            print(sm)
            cm = complex_math(sm)
            maths += cm
        print("%d"%(maths))
    

main()

File: 2019/day1c/day1.c
#include<stdio.h>
#include<math.h>
#include"day1.h"

//this will do the math to solve the problem
int simple_math( int input){
	double answer;
	// first we are going to divided input by three
	answer = (double) input / 3;
	// then we get the floor of the answer 
	answer = floor(answer);
	// then we subtract two;
	answer = answer - 2;
	//return the value
	return (int) answer;
}
int complex_math(int input) {
	// function was created by sarahfossheim and encode into c 
	int total_fuel;
	total_fuel =0;

	while (1){
		//As long as the fuel is positive, keep calculationg
		input = simple_math(input);
		if (input > 0){
			total_fuel += input; 
		} else {
			break;
		}
	}

	return  total_fuel;
}


int main(){
	FILE *f = fopen("main.txt", "r");
	int number = 0;
	int sum = 0;

	while (fscanf(f,"%d\n", &number)>0)
	{
		sum1+=simple_math(number);
		sum+=complex_math(number);
	}
	
	fclose(f);
	printf("sm: %d\n cm: %d\n",sum);

	return 0;
}

